### PR TITLE
Python 3 compatibility: Use relative imports

### PR DESCRIPTION
**Pull request by SaschaNaz**
Python 3 forces `import module` to only import global modules from system path.

```py
# Python 2 only
import my_local_module
from my_local_module import my_variable
```

```py
# Python 2 and 3
from . import my_local_module
from .my_local_module import my_variable
```

### CODE REVIEW

File: tools/ctor_evaller.py [hunk 1]
```diff
@@ -5,8 +5,11 @@
 '''
 
 import os, sys, json, subprocess, time
-import shared, js_optimizer
-from tempfiles import try_delete
+
+sys.path.insert(1, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
```
Comment (kripken) [inline tools/ctor_evaller.py]: why is this needed?

Comment (SaschaNaz) [reply tools/ctor_evaller.py]: This is to access `tools` package correctly by adding proper system path. `emcc.py` can access `tools` because the root path is already automatically added to system path, but not in this case as `ctor_evaller` is inside `tools` directory.

It cannot access the package by `from . import shared` because 1) `from .` does not load any package 2) so is only allowed to package internal modules not scripts.

Related SO post: https://stackoverflow.com/questions/11536764/how-to-fix-attempted-relative-import-in-non-package-even-with-init-py

Comment (SaschaNaz) [reply tools/ctor_evaller.py]: `import` on Python 2 is a magic and it does all the path and relative import things implicitly. Not in Python 3.

Comment (SaschaNaz) [reply tools/ctor_evaller.py]: https://github.com/kripken/emscripten/blob/incoming/tests/runner.py#L25

An existing example from the current code.

Comment (kripken) [reply tools/ctor_evaller.py]: Got it, thanks.

File: tools/asm_module.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 from __future__ import print_function
 import sys, re, itertools
 
-import shared, js_optimizer
+from . import shared, js_optimizer
```
Comment (kripken) [inline tools/asm_module.py]: wait, then why does this work?

Comment (SaschaNaz) [reply tools/asm_module.py]: Because `tools/asm_module.py` is always imported as a module by script and never directly runs as a script. (In other words, we never call `python tools/asm_module.py`.)

Modules support `from .` but scripts don't.

Comment (kripken) [reply tools/asm_module.py]: Ok, thanks. Now I really do get it :)

**Comment by sbc100** [comment]
Did this break this test:
```
Traceback (most recent call last):
  File "/b/build/slave/linux/build/src/src/work/wasm-install/emscripten/tools/webidl_binder.py", line 11, in <module>
    from . import shared
ValueError: Attempted relative import in non-package
test_webidl (test_core.binaryen2) ... FAIL
```

**Comment by kripken** [comment]
Thanks, yeah. We happen to not run those tests on CI so it was missed :(

Fix in #5712
